sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_client.hpp

```diff
@@ -73,6 +73,13 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
         std::shared_ptr<Core::Credentials::TokenCredential const> credential,
         KeyClientOptions options = KeyClientOptions());
 
+    /**
+     * @brief Construct a new Key Client object from another key client.
+     *
+     * @param keyClient An existing key vault key client.
+     */
+    explicit KeyClient(KeyClient const* keyClient) : m_pipeline(keyClient->m_pipeline) {}
```

Comment:
Passing pointer looks new to me. We used to use const ref.

Reply:
This is just to be able to use `this` directly from a client's operation like
client.Foo() {
   // Create a client based on myself
   auto client = Client(this);
}

Using const& the code would be 
client.Foo() {
   // Create a client based on myself
   auto client = Client(*this);
}

Reply:
Think I can use const& to be consistent